Framework/CurveFitting/src/Algorithms/SplineSmoothing.cpp

```diff
@@ -109,16 +113,21 @@ void SplineSmoothing::exec() {
  *
  * @param index :: index of the spectrum to smooth
  */
-void SplineSmoothing::smoothSpectrum(int index) {
+void SplineSmoothing::smoothSpectrum(const int index) {
   m_cspline = boost::make_shared<BSpline>();
   m_cspline->setAttributeValue("Uniform", false);
 
   // choose some smoothing points from input workspace
   selectSmoothingPoints(m_inputWorkspacePointData, index);
   performAdditionalFitting(m_inputWorkspacePointData, index);
 
-  // compare the data set against our spline
-  m_outputWorkspace->setX(index, m_inputWorkspace->refX(index));
```

Comment:
I think you can avoid all the new stuff you added and just replace the old line by this:

```
m_outputWorkspace->setPoints(index, m_inputWorkspace->points(index));
```